Sources/tuist/TuistCLI.swift

```diff
@@ -45,8 +51,18 @@ private enum TuistCLI {
         context.logger = Logger(label: "dev.tuist.cli", factory: loggerHandler)
 
         try await ServiceContext.withValue(context) {
-            try await TuistCommand.main()
+            try await TuistCommand.main(logFilePath: logFilePath)
+        }
+    }
+
+    private static func touchLogFile() async throws -> Path.AbsolutePath {
+        let fileSystem = FileSystem()
+        let logFilePath = Environment.shared.stateDirectory.appending(components: ["logs", "\(UUID().uuidString).log"])
+        if !(try await fileSystem.exists(logFilePath.parentDirectory)) {
+            try await fileSystem.makeDirectory(at: logFilePath.parentDirectory)
         }
+        try await fileSystem.touch(logFilePath)
+        return logFilePath
     }
```

Comment:
It's better to keep the `TuistCLI` as thin as possible. This feels like it could be a part of `LogsCleaner` if we rename it to something like `LogsService`?